pandas/src/generate_code.py

```diff
@@ -2286,6 +2292,45 @@ def generate_put_template(template, use_ints=True, use_floats=True,
         output.write(func)
     return output.getvalue()
 
+def generate_put_selection_template(template, use_ints=True, use_floats=True,
+                                    use_objects=False, use_datelikes=False):
+    floats_list = [
+        ('float64', 'float64_t', 'float64_t', 'np.float64', 'nan', 'np.inf'),
+        ('float32', 'float32_t', 'float32_t', 'np.float32', 'nan', 'np.inf'),
+    ]
+    ints_list = [
+        ('int8',  'int8_t',  'int8_t', 'np.int8', 'iNaT', _int8_max),
```

Comment:
The min and max ops use this. I wanted to make sure that whatever dtype went in was the same dtype that came out, and without using the max for the same dtype, the cython functions produce erroneous results.